htdocs/core/class/html.form.class.php

```diff
@@ -4055,6 +4055,9 @@ public function select_produits_fournisseurs_list($socid, $selected = '', $htmln
 					'outvallabel' => &$outvallabel,
 					'outarrayentry' => &$outarrayentry
 				);
+				if (is_null($objp->fk_soc)) {
```

Comment:
Obj var represent result of a sql, content must not be modified. To propagate an information to the trigger, you can complete the $parameters array by adding an entry 'socid' => $socid for example